src/compiler/parser.ts

```diff
@@ -6471,7 +6471,23 @@ namespace ts {
                     return finishNode(result, end);
                 }
 
+                function isNextNonwhitespaceTokenEndOfFile(): boolean {
+                    nextJSDocToken();
+                    if (token() === SyntaxKind.EndOfFileToken) {
+                        return true;
+                    }
+                    if (token() === SyntaxKind.WhitespaceTrivia || token() === SyntaxKind.NewLineTrivia) {
+                        return lookAhead(isNextNonwhitespaceTokenEndOfFile); // We must use infinte lookahead, as there could be any number of newlines :(

```

Comment:
typo:infinite

Reply:
why do we need lookahead inside of lookahead? Is there a way to get around that?

Reply:
Yeah, I think you should be able to `return isNextNonwhitespaceTokenEndOfFile()` instead

Reply:
Even just recursion for each line/whitespace seems suboptimal, but preferable, and it not like you'll have thousands of blank lines in a source file (hopefully!)

Reply:
Yeah. It's just a little unfortunate that _jsdoc parsing_ needs infinite lookahead - previously only arrow functions needed that.